File: Java/BridgePattern/src/resources/AlbumResource.java
package resources;

import models.Album;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 13-06-2021
 * Time: 08:53 PM
 */
public class AlbumResource implements IResource{

    Album album;

    public AlbumResource(Album album) {
        this.album = album;
    }

    @Override
    public String snippet() {
        return "";
    }

    @Override
    public String title() {
        return album.getAlbumName();
    }

    @Override
    public String image() {
        return album.getAlbumImage();
    }

    @Override
    public String url() {
        return album.getAlbumLink();
    }
}

File: Kotlin/BridgePattern/src/resources/AlbumResource.kt
package resources

import models.Album

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 13-06-2021
 *   Time: 09:33 PM
 *
 */
class AlbumResource(private val album: Album): IResource {

    override fun snippet(): String {
        return ""
    }

    override fun title(): String {
        return album.albumName
    }

    override fun image(): String {
        return album.albumImage
    }

    override fun url(): String {
        return album.albumLink
    }
}